src/sentry/reprocessing2.py

```diff
@@ -202,16 +213,46 @@ def mark_event_reprocessed(data):
     _get_sync_redis_client().decr(key)
 
 
-def start_group_reprocessing(project_id, group_id, max_events=None):
-    from sentry.models.group import Group, GroupStatus
-    from sentry.models.grouphash import GroupHash
+def start_group_reprocessing(project_id, group_id, max_events=None, acting_user_id=None):
     from django.db import transaction
 
     with transaction.atomic():
-        Group.objects.filter(id=group_id).update(status=GroupStatus.REPROCESSING)
-        # Remove all grouphashes such that new events get sorted into a
-        # different group.
-        GroupHash.objects.filter(group_id=group_id).delete()
+        group = models.Group.objects.get(id=group_id)
+        transferred_group_attrs = {}
+        for attr in TRANSFER_GROUP_ATTRS:
+            transferred_group_attrs[attr] = getattr(group, attr)
+
+        group.status = models.GroupStatus.REPROCESSING
+        group.short_id = None  # satisfy unique constraint
```

Comment:
Note that this relies on Postgres specific behavior, which is fine now that we only support postgres.

Reply:
what's the pg specific behavior? I can add a comment

Reply:
That NULL is excluded from unique constraints. AFAIK, MSSQL does enforce this, but I might be wrong.

Reply:
Ah yes. I added a comment. This slipped past me.